packages/fury-adapter-oas3-parser/lib/parser/oas/parseServerVariableObject.js

```diff
@@ -0,0 +1,50 @@
+const R = require('ramda');
+const { createWarning } = require('../../elements');
+const {
+  createInvalidMemberWarning,
+} = require('../annotations');
+const {
+  isObject, hasKey, isExtension, getValue,
+} = require('../../predicates');
+const parseObject = require('../parseObject');
+const parseString = require('../parseString');
+const parseEnum = require('../parseEnum');
+const pipeParseResult = require('../../pipeParseResult');
+
+const name = 'Server Variable Object';
+const requiredKeys = ['default'];
+
+const parseMember = context => R.cond([
+  [hasKey('enum'), R.compose(parseEnum(context, name), getValue)],
```

Comment:
The OAS spec specifies that `enum` for a Server Variable Object should be an array of string which this implementation is not enforcing. This is not the same rule for Schema Object. I think we will need to make another `parseEnum` specific for Server Variable Object to warn and discard (or coerce) values which are not strings in the enum. It could make use of the shared `parseEnum` but add additional checks for the contents of the array.

That said, it probably isn't worth the effort I think it will be simpler to just handle the enums here because `parseArray` accepts a sub-parser (for example the string parser):

```js
const parseEnum = pipeParseResult(
  parseArray(context, `${name}' 'enum`, parseString(...)),
  (array) => {
    const element = new context.namespace.elements.Enum(array);
    element.enumerations.forEach(value => value.attributes.set('typeAttributes', ['fixed']));
    return element;
  }
);
```

We should have a test case for parseServerVariableObject's when the enum contains non strings. For example if the source was:

```js
['a', 1, 'b', true]
```

The resulting enum should be of `'a'` and `'b'`, and there should be two warnings, one for the value `1` and `true`.